src/rust/plugin-registry/src/server/get_plugin_health.rs

```diff
@@ -53,3 +60,18 @@ async fn query_nomad_for_health(
         )),
     }
 }
+
+/// Given a plugin-id, query Consul to see if the service
+/// `plugin-${plugin_id}` is healthy.
+async fn query_consul_for_service_health(
+    plugin_id: uuid::Uuid,
+) -> Result<PluginHealthStatus, PluginRegistryServiceError> {
+    let service_name = format!("plugin-{plugin_id}");
+    let consul_client = ConsulClient::new(ConsulClientConfig::parse());
+    let service_health = consul_client.check_health(service_name).await?;
+    tracing::info!(message = "health result", service_health =? service_health);
+    match service_health.all_passing() {
```

Comment:
What if one of them is e.g. `"critical"`? Wouldn't that map better to `PluginHealthStatus::Dead`? This mapping might be better expressed as an `impl From<models::CheckHealthResponse> for PluginHealthStatus`

Reply:
fixed